Cyware Solution - Initial Addition

AashiqRamachandran opened the pull request:
Change(s):
   - Initial addition of Cyware Labs connectors

   Reason for Change(s):
   - Initial addition

   Version Updated:
   - Initial 1.0.0 version

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Need assistance (Can I get additional clarification around packaging the solutions if further packaging is needed?)

Thanks!

v-atulyadav commented on Solutions/Cyware/Hunting queries/UnusualNetworkConnectionsToRareExternalDomains.yaml:
```diff
@@ -0,0 +1,22 @@
+'DeviceProcessEvents'
+]
+tactics:
+  - Command And Control
```
Hi @AashiqRamachandran, tactics names MUST NOT have any spaces. It should be `CommandAndControl`. Also relevantTechniques missing for this tactic.

v-atulyadav commented on Solutions/Cyware/Hunting queries/MatchCywareIntelWatchlistItemsWithCommonLogs.yaml:
```diff
@@ -0,0 +1,26 @@
+'DeviceProcessEvents'
+]
+tactics:
+  - Command And Control
```
Hi @AashiqRamachandran, tactics names MUST NOT have any spaces. It should be `CommandAndControl`. Also relevantTechniques missing for this tactic.

v-atulyadav commented on Solutions/Cyware/Hunting queries/DetectingSuspiciousPowerShellCommandExecutions.yaml:
```diff
@@ -0,0 +1,34 @@
+| project TimeGenerated, DeviceName, InitiatingProcessCommandLine, ProcessCommandLine, SearchKey, LastUpdatedTimeUTC
+
+version: 1.0.0
+metadata:
```
Hi @AashiqRamachandran, Metadata is not required for hunting queries within solution.

v-atulyadav commented on Solutions/Cyware/Hunting queries/MatchCywareIntelWatchlistItemsWithCommonLogs.yaml:
```diff
@@ -0,0 +1,26 @@
+) on $left.DestinationIP == $right.SearchKey
+| project TimeGenerated, SourceIP, DestinationIP, DestinationHostName, Activity, ApplicationProtocol, SearchKey=SearchKey, LastUpdatedTimeUTC
+version: 1.0.0
+metadata:
```
Hi @AashiqRamachandran, Metadata is not required for hunting queries within solution

v-atulyadav commented on Solutions/Cyware/Hunting queries/UnusualNetworkConnectionsToRareExternalDomains.yaml:
```diff
@@ -0,0 +1,22 @@
+) on $left.RemoteUrl == $right.SearchKey
+| project TimeGenerated, RemoteUrl, DeviceName, SearchKey=SearchKey, LastUpdatedTimeUTC
+version: 1.0.0
+metadata:
```
Hi @AashiqRamachandran, Metadata is not required for hunting queries within solution

v-atulyadav commented on Solutions/Cyware/Playbooks/Send_Incident_To_Cyware_Orchestrate/README.md:
```diff
@@ -0,0 +1,77 @@
+# Send Incident to Cyware Orchestrate
+
+This Azure Logic App template is designed to integrate Azure Sentinel incidents with Cyware Orchestrate, enabling automated incident response actions. When an Azure Sentinel alert triggers, this Logic App automatically sends the incident details to Cyware Orchestrate.
```
Hi @AashiqRamachandran,

Please replace `Azure Sentinel` text with `Microsoft Sentinel` in all occurrences. Thanks

v-atulyadav commented:
Hi @AashiqRamachandran,
1. Playbook file azuredeploy.json has missing metadata, please add, below playbook sample is for your reference.
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Recorded%20Future/Playbooks/ThreatHunting/RecordedFuture-ActorThreatHunt-IndicatorImport/azuredeploy.json
2. Also Images folder missing in playbook please add working png images into that. Below for your reference
https://github.com/Azure/Azure-Sentinel/tree/master/Playbooks/ADX-Health-Playbook

AashiqRamachandran commented:
Hi @v-atulyadav,

Many thanks for the prompt review! I've added all the changes requested

- Removed Metadata in hunting queries
- Fix tactics and techniques referenced in hunting queries
- Fix azuredeploy.json with metadata
- Add images to README.md file for the Sentinel playbooks 

Please do let me know any other changes that may be required!

v-atulyadav commented:
Hi @AashiqRamachandran,
I have resolved the validations by modifying the files. Please find the attached zip files and follow the guidelines below.
[Playbook.zip](https://github.com/Azure/Azure-Sentinel/files/14511251/Playbook.zip)
[Hunting.zip](https://github.com/Azure/Azure-Sentinel/files/14511252/Hunting.zip)

1. Extract Hunting.zip and replace these files into hunting folder
2. Extract playbook.zip and replace this file into below location.
<img width="574" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/104008048/1f8b39e8-db53-4449-ac07-fb90b8791a8e">

v-atulyadav commented:
Hi @AashiqRamachandran,
1. Please repackage this solution with the help of below mentioned link.
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md
2. Also check below link to add Release Notes in solution
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md
3. Please provide me an access to your branch so I will be able to commit if required.

v-atulyadav commented:
Hi @AashiqRamachandran,
We are waiting for your updates. Thanks

AashiqRamachandran commented:
Hi @v-atulyadav ,

Yes. We are currently packaging the solution as per the document attached above. We are expecting to push the updated code in the next few hours. Will let you know once pushed

AashiqRamachandran commented:
Hi @v-atulyadav,

We have implemented all the required changes, and the content has also been pushed. Please let me know any additional fixes to implement.

Looking forward to hearing back

v-atulyadav commented:
Hi @AashiqRamachandran,
Thank you for repackaging this solution, I will review this again. Coud you please modify release note version it should be 3.0.0 instead of 1.0.0.

v-atulyadav commented:
Hi @AashiqRamachandran,
Logo validations are failing. I have modified the logo please extract below file and replace. 
[cyware-logo.zip](https://github.com/Azure/Azure-Sentinel/files/14646179/cyware-logo.zip)

Also modify RelaseNote version to 3.0.0
<img width="541" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/104008048/5b0adb71-45e5-4af2-b60d-a01fb6dad662">

AashiqRamachandran commented:
Hi @v-atulyadav ,

I've added the fixes. A small doubt on the version tho. This is the first time we are developing a sentinel integration. Will the version still be 3.0.0? (The changes are pushed regardless)

Looking forward to hearing back!

v-atulyadav commented:
> Hi @v-atulyadav ,
> 
> I've added the fixes. A small doubt on the version tho. This is the first time we are developing a sentinel integration. Will the version still be 3.0.0? (The changes are pushed regardless)
> 
> Looking forward to hearing back!

Yes, we are using v3 tool for repackaging hence initial version should be 3.0.0.

AashiqRamachandran commented:
Understood @v-atulyadav.

Please let me know any additional changes

AashiqRamachandran commented:
Hi @v-atulyadav ,

I've renamed the files as recommended. Also checking on adding you as a contributor to our repo to enable you to make edits.

Do let me know if anything else is needed!

AashiqRamachandran commented:
Hi @v-atulyadav ,

I see that the Pipeline checks have all passed (17/17). Please do let us know of next steps and changes that may be required!

Looking forward to hearing back

v-atulyadav commented:
Hi @AashiqRamachandran,
We still need a few changes, as we are unable to view the playbook after deployment. I am working on this issue. Meanwhile, could you please provide me with access so that I can commit a few changes and investigate the issue?. Thanks

v-atulyadav commented:
Hi @AashiqRamachandran,
A. Please provide me an access for your branch. 
B. 2 hunting queries have the same GUID, which you need to update for 1 file and repackage. Thanks
<img width="796" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/104008048/a5a929ed-ba45-4a2a-99a5-f7eba2cfb3c0">

v-atulyadav commented:
Hi @AashiqRamachandran,
We are waiting for your response on above. Thanks

AashiqRamachandran commented:
Hi @v-atulyadav ,

Sorry for the late reply. Was OOO due to an emergency. Yes. Will add the changes, and let will keep you posted on the repo access ASAP!

ashwinkarkalahegde commented:
> Hi @AashiqRamachandran, We are waiting for your response on above. Thanks

Hi Atul, I added you as a collaborator on the repo. Currently you have read access. If you need further access let me know.

v-atulyadav commented:
Sure @AashiqRamachandran, Thanks.

v-atulyadav commented:
Hi @AashiqRamachandran,
I am still not able to push anything into your branch. The below screenshot is for your reference. So, I am requesting that please check the above comments and act accordingly. Thanks
<img width="713" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/104008048/53661206-b9ce-4c82-bcac-ba7013ebc85c">

AashiqRamachandran commented:
Hi @v-atulyadav - Looking into this. 

@ashwinkarkalahegde FYI

AashiqRamachandran commented:
Hi @v-atulyadav ,

From investigating the issue, looks like your invite had expired. Resent the invite as well! Please let us know any additional changes required!

![Screenshot 2024-04-19 at 12 28 32 PM](https://github.com/Azure/Azure-Sentinel/assets/49115233/e9fc51da-7e81-49bd-8d07-b10c5945da17)
![Screenshot 2024-04-19 at 12 29 37 PM](https://github.com/Azure/Azure-Sentinel/assets/49115233/4123c38e-54d3-4ac3-bfd3-ac1aaea0c830)

v-atulyadav commented:
Thanks @AashiqRamachandran.

v-atulyadav commented:
Sure @AashiqRamachandran. Thanks

v-atulyadav commented:
Hi @AashiqRamachandran,
I am facing an issue with PlayBook after the maintemplate custom deployment; PlayBook is unable to load in the PlayBook automation tab. It seems the issue is with the playbook schema. Could you please refer to or compare the below sample playbook and make the necessary changes accordingly?. Thanks
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Recorded%20Future/Playbooks/Alerts/RecordedFuture-Alert-Importer/azuredeploy.json

v-atulyadav commented:
Hi @AashiqRamachandran,
Please act on above comment. Thanks

v-atulyadav commented:
Hi @AashiqRamachandran,
We are waiting for your reply. Thanks

AashiqRamachandran commented:
I think I see the issue and am fixing it @v-atulyadav . Will push the newer code by tomorrow EOD IST

v-atulyadav commented:
Thanks @AashiqRamachandran.

v-atulyadav commented:
Hi @AashiqRamachandran,
Waiting for your response. Thanks

v-atulyadav commented:
Hi @AashiqRamachandran,
We are waiting for your reply. Thanks

AashiqRamachandran commented:
Hi @v-atulyadav ,

Quick question. Are the rules part of the solution fine? Or any changes are required there as well?

v-atulyadav commented:
> Hi @v-atulyadav ,
> 
> Quick question. Are the rules part of the solution fine? Or any changes are required there as well?

Rule part of the solution is fine.

AashiqRamachandran commented:
Sounds good. Another question from there. Do we have to create a Marketplace listing for the rules as well as part of the solution?

v-atulyadav commented:
> Sounds good. Another question from there. Do we have to create a Marketplace listing for the rules as well as part of the solution?

Yes you need to create marketplace offer for this solution after PR merge.
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/README.md

AashiqRamachandran commented:
@v-atulyadav - I Have regenerated the azuredeploy.json from using the MS published tool Playbook_ARM_Template_Generator instead of manually writing it like how we did it last time. 

Do I have to regenerate the zip for you to test, or does this work?

AashiqRamachandran commented:
FYI @v-atulyadav - While running the tool, we persistently ran into an error while the browser opened for signing into Azure. The command that fixed it was - Update-AzConfig - EnableLoginByWam $false

v-atulyadav commented:
> @v-atulyadav - I Have regenerated the azuredeploy.json from using the MS published tool Playbook_ARM_Template_Generator instead of manually writing it like how we did it last time.
> 
> Do I have to regenerate the zip for you to test, or does this work?

You need to regenerate the zip file and also need repackaging. Thanks

v-atulyadav commented:
Hi @AashiqRamachandran,
Any update on above suggestions? Thanks

AashiqRamachandran commented:
We’re repackaging the solution Atul. Running into some issues while
packaging

Aashiq Ramachandran

Contact:*+91 8248743303*
LinkedIn:* linkedin.com/in/aashiq-ramachandran
<http://linkedin.com/in/aashiq-ramachandran>*


On Thu, 6 Jun 2024 at 12:21 PM, v-atulyadav ***@***.***>
wrote:

> Hi @AashiqRamachandran <https://github.com/AashiqRamachandran>,
> Any update on above suggestions? Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/Azure-Sentinel/pull/10077#issuecomment-2151534693>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALWXAYMGAZ2WHSR3OXGTL5TZGABH7AVCNFSM6AAAAABEDUW7IGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNJRGUZTINRZGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

v-atulyadav commented:
Hi @AashiqRamachandran,
I will take care of the packaging and let you know if there is anything else we require from your side. Thanks

AashiqRamachandran commented:
Thank you @v-atulyadav . Please let me know any further changes

v-atulyadav commented:
Sure @AashiqRamachandran. Thanks

v-atulyadav commented:
Hi @AashiqRamachandran,
Earlier, we added playbook metadata, but now a few properties are missing. Please add the below-marked property values. Thanks
<img width="377" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/104008048/15bbaa28-3fbc-4fae-a243-3551b6e46760">

AashiqRamachandran commented:
I can add the metadata now Atul

Aashiq Ramachandran

Contact:*+91 8248743303*
LinkedIn:* linkedin.com/in/aashiq-ramachandran
<http://linkedin.com/in/aashiq-ramachandran>*


On Tue, 18 Jun 2024 at 1:47 PM, v-atulyadav ***@***.***>
wrote:

> Hi @AashiqRamachandran <https://github.com/AashiqRamachandran>,
> Earlier, we added playbook metadata, but now a few properties are missing.
> Please add the below-marked property values. Thanks
> image.png (view on web)
> <https://github.com/Azure/Azure-Sentinel/assets/104008048/15bbaa28-3fbc-4fae-a243-3551b6e46760>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/Azure-Sentinel/pull/10077#issuecomment-2175491378>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALWXAYMJZRP726GPKQH4GTLZH7UK7AVCNFSM6AAAAABEDUW7IGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNZVGQ4TCMZXHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

AashiqRamachandran commented:
@v-atulyadav - I have re-added the meta content. Please let me know anything else required!

v-atulyadav commented:
Hi @AashiqRamachandran,
Validations are failing please check. Thanks

AashiqRamachandran commented:
Looking into this Atul. Give me a few mins

Aashiq Ramachandran

Contact:*+91 8248743303*
LinkedIn:* linkedin.com/in/aashiq-ramachandran
<http://linkedin.com/in/aashiq-ramachandran>*


On Mon, 24 Jun 2024 at 12:24 PM, v-atulyadav ***@***.***>
wrote:

> Hi @AashiqRamachandran <https://github.com/AashiqRamachandran>,
> Validations are failing please check. Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Azure/Azure-Sentinel/pull/10077#issuecomment-2185747379>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALWXAYPH3JYGN3OWYCOKUPTZI67DPAVCNFSM6AAAAABEDUW7IGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOBVG42DOMZXHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

AashiqRamachandran commented:
Seemed like a JSON file error @v-atulyadav. Fixed and repushed.

AashiqRamachandran commented:
All pipelines are working now @v-atulyadav. Please let me know next steps!

AashiqRamachandran commented:
Hey @v-atulyadav - Quickly wanted to check in on if there were any action items to be completed from my end!

v-atulyadav commented:
Hi @AashiqRamachandran,
I will repackage this solution and let you know if anything is needed from your side. Thanks

AashiqRamachandran commented:
Sure @v-atulyadav. Please do let me know in case anything else from my side is needed, Thanks!

AashiqRamachandran commented:
Hi @v-atulyadav - Quickly wanted to check in on anything required from our end towards getting our solution merged!

v-atulyadav commented:
Hi @AashiqRamachandran,
Not now, as we have already repackaged your change. We are reviewing this and will let you know about it.

AashiqRamachandran commented:
Thank you @v-atulyadav! We are super excited to merge and am looking forward to hearing back!

AashiqRamachandran commented:
Hi @v-atulyadav, hope all is well on your end! I quickly wanted to check in on the internal validations you mentioned and if any changes/action items are required from our end!
